components/game/lib.ts

```diff
@@ -128,22 +137,30 @@ export class Game {
             currentCount = 1;
           }
           const won = currentPlayer && currentCount >= WIN_LEN;
-          if (won) return currentPlayer as Player;
+          if (won) return currentPlayer as PlayerName;
         }
       }
     }
     return null;
   }
 
-  nextPlayer(): Player {
+  nextPlayer(): PlayerName {
     return this.field.flat().filter((v) => v !== 0).length % 2 === 0
-      ? (1 as Player)
-      : (2 as Player);
+      ? (this.playerOneName as PlayerName)
+      : (this.playerTwoName as PlayerName);
   }
 
   areCoordsValid(coords: Coords) {
     return !!this.possibleCoords().find(
       (cs) => cs.x === coords.x && cs.y === coords.y
     );
   }
+
+  setPlayerName(playerName: PlayerName, player: number) {
```

Comment:
minor note, arg order should be player -> playerName, same as with `Map.set(key, value)`